### ISSUE TITLE
TypeError: Cannot read property 'fetch' of null

### ISSUE THREAD

**Issue by 7twin** [jsdom/jsdom#2701]
### Basic info:

- **Node.js version:** v12.10.0
- **jsdom version:** 15.2.0

### Possible Solution

By patching in the previously suggested PR: https://github.com/jsdom/jsdom/pull/2510 fixes this issue too, would it be possibly better to merge that and check for consequence (`_resourceLoader === null`), instead of checking for cause? in the end it shouldn't matter what causes the `_resourceLoader` to default to `null` here: https://github.com/jsdom/jsdom/blob/master/lib/jsdom/browser/resources/per-document-resource-loader.js#L9 or why is there a reason it would matter? further down the fetch function it returns null anyway in other cases, so it seems safe to return null if `_resourceLoader` was defaulted to null too, instead of attempting to call `.fetch()` on it?

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

doc = new JSDOM().window.document.implementation.createHTMLDocument();
htmlElement = doc.getElementsByTagName('html')[0];
htmlElement.innerHTML = `
    <img src="https://test.com/test.png">
`;
```

### Error

```
xxx/node_modules/jsdom/lib/jsdom/browser/resources/per-document-resource-loader.js:18
    const request = this._resourceLoader.fetch(url, {
                                         ^

TypeError: Cannot read property 'fetch' of null
    at PerDocumentResourceLoader.fetch (xxx/node_modules/jsdom/lib/jsdom/browser/resources/per-document-resource-loader.js:18:42)
    at HTMLImageElementImpl._attrModified (xxx/node_modules/jsdom/lib/jsdom/living/nodes/HTMLImageElement-impl.js:39:36)
    at Object.exports.appendAttribute (xxx/node_modules/jsdom/lib/jsdom/living/attributes.js:62:11)
    at Object.exports.setAttributeValue (xxx/node_modules/jsdom/lib/jsdom/living/attributes.js:218:13)
    at JSDOMParse5Adapter.adoptAttributes (xxx/node_modules/jsdom/lib/jsdom/browser/parser/html.js:152:18)
    at JSDOMParse5Adapter.createElement (xxx/node_modules/jsdom/lib/jsdom/browser/parser/html.js:73:10)
    at Parser._appendElement (xxx/node_modules/parse5/lib/parser/index.js:553:42)
    at areaStartTagInBody (xxx/node_modules/parse5/lib/parser/index.js:1519:7)
    at Object.startTagInBody [as START_TAG_TOKEN] (xxx/node_modules/parse5/lib/parser/index.js:1725:17)
    at Parser._processToken (xxx/node_modules/parse5/lib/parser/index.js:657:55)
```

**Comment by domenic** [comment]
Thanks for the report!

We shouldn't be making quick-fix patches that cover over the real problem. In this case the real problem is that documents without browsing contexts (e.g. those created by `document.implement.createHTMLDocument()`) should never be sending out requests in the first place. That's a bug, and we shouldn't paper over it by letting them send out requests, but when they fail, hiding the failure.

In other words, you can also "fix" lots of bugs by putting a big `try { ... } catch { /* do nothing */ }` around your program, but we don't want to accept patches that do that :).

We'll work on fixing the actual source of this issue ASAP.

**Comment by 7twin** [comment]
@domenic thanks, makes sense, just that it's been coming back multiple times, breaking basic function and the code does default to "null" so eventually in some edge-case it will end up in .fetch() being called on null - or is that on purpose to have it break and find the "real" issue each time?

**Comment by domenic** [comment]
Yep, exactly. But no, I don't think hiding bugs with warnings is a good idea.

**Comment by 7twin** [comment]
@domenic I see, I suppose another option would be then to expand the tests to cover more scenarios to catch those "basic" tag issues like with <link> and <img> (incl. others that trigger fetch, I assume all that have a src attribute) and just keep it failing if it is called on null
